l1-contracts/src/core/messagebridge/Outbox.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 Aztec Labs.
+pragma solidity >=0.8.18;
+
+import {MessageBox} from "./MessageBox.sol";
+
+/**
+ * @title Outbox
+ * @author Aztec Labs
+ * @notice Lives on L1 and is used to consume L2 -> L1 messages. Messages are inserted by the rollup contract
+ * and will be consumed by the portal contracts.
+ */
+contract Outbox is MessageBox {
+  error Outbox__Unauthorized();
+  error Outbox__WrongChainId();
+
+  /**
+   * @dev  struct for sending messages from L2 to L1
+   * @param sender - The sender of the message
+   * @param recipient - The recipient of the message
+   * @param content - The content of the message (application specific) padded to bytes32 or hashed if larger.
+   */
+  struct L2ToL1Msg {
+    L2Actor sender;
+    L1Actor recipient;
+    bytes32 content;
+  }
+
+  // to make it easier for portal to know when to consume the message.
+  event MessageAdded(
+    bytes32 indexed entryKey,
+    bytes32 indexed sender,
+    address indexed recipient,
+    uint256 senderVersion,
+    uint256 recipientChainId,
+    bytes32 content
+  );
+
+  event MessageConsumed(bytes32 indexed entryKey, address indexed recipient);
```

Comment:
When consuming, you will have the stuff that you put in `MessageAdded`, might be useful for data analysis etc to put in in there instead.